src/modalities/models/coca/multi_modal_decoder.py

```diff
@@ -0,0 +1,116 @@
+from functools import partial
+from typing import Dict
+
+import torch
+import xformers.ops as xops
+from torch import nn
+
+from modalities.models.gpt2.gpt2_model import ActivationType, LayerNorm
+from modalities.models.model import NNModel
+from modalities.nn.attention import AttentionConfig, AttentionType, MultiHeadAttention
+from modalities.nn.mlp import MLP
+
+
+class MultiModalBlock(nn.Module):

```

Comment:
`MultiModalBlock` is also used for the pure `TextDecoder`. Hence, I think it would be better to call it `DecoderBlock`. In that case, it might also be good to remove the default values for the last two arguments `attention_type` and `with_context`, so that they always have to be explicitly specified no matter whether the `DecoderBlock` is used in the pure text or multimodal context.

Reply:
Maybe the name `TransformerBlock` would be even better

Reply:
https://github.com/Modalities/modalities/pull/16/commits/be9e01d2156f7fc00f681286c9f202bb77afd5ec